components/brave_wallet/browser/zcash/create_shield_all_transaction_task.cc

```diff
@@ -0,0 +1,251 @@
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace brave_wallet {
+
+namespace {
+
+std::unique_ptr<OrchardManager> ApplyOrchardSignatures(
+    std::unique_ptr<OrchardManager> orchard_manager,
+    std::array<uint8_t, kZCashDigestSize> sighash) {
+  return std::move(orchard_manager)->ApplySignature(sighash);
+}
+
+}  // namespace
+
+CreateShieldAllTransactionTask::CreateShieldAllTransactionTask(
+    ZCashWalletService* zcash_wallet_service,
+    const std::string& chain_id,
+    const mojom::AccountIdPtr& account_id,
+    ZCashWalletService::CreateTransactionCallback callback,
+    std::optional<uint64_t> random_seed_for_testing)
+    : zcash_wallet_service_(zcash_wallet_service),
+      chain_id_(chain_id),
+      account_id_(account_id.Clone()),
+      callback_(std::move(callback)),
+      random_seed_for_testing_(random_seed_for_testing) {}
+
+CreateShieldAllTransactionTask::~CreateShieldAllTransactionTask() = default;
+
+void CreateShieldAllTransactionTask::Iterate() {
+  if (error_) {
+    std::move(callback_).Run(base::unexpected(error_.value()));
+    return;
+  }
+
+  if (!tree_state_) {
+    GetTreeState();
+    return;
+  }
+
+  if (!utxo_map_) {
+    GetAllUtxos();
+    return;
+  }
+
+  if (!chain_height_) {
+    GetChainHeight();
+    return;
+  }
+
+  if (!transaction_ && !CreateTransaction()) {
+    std::move(callback_).Run(base::unexpected(error_.value()));
+    return;
+  }
+
+  // Signing of transparent part is final step
+  if (!transaction_->IsTransparentPartSigned()) {
+    CompleteTransaction();
+    return;
+  }
+
+  std::move(callback_).Run(std::move(transaction_.value()));
+}
+
+bool CreateShieldAllTransactionTask::CreateTransaction() {
+  CHECK(utxo_map_);
+  CHECK(chain_height_);
+
+  ZCashTransaction zcash_transaction;
+
+  // Pick inputs
+  std::vector<ZCashTransaction::TxInput> all_inputs;
+  for (const auto& item : utxo_map_.value()) {
+    for (const auto& utxo : item.second) {
+      if (!utxo) {
+        error_ = l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR);
+        return false;
+      }
+      if (auto input =
+              ZCashTransaction::TxInput::FromRpcUtxo(item.first, *utxo)) {
+        all_inputs.emplace_back(std::move(*input));
+      }
+    }
+  }
+
+  zcash_transaction.transparent_part().inputs = std::move(all_inputs);
+  // TODO(cypt4): Calculate orchard actions count
+  zcash_transaction.set_fee(CalculateZCashTxFee(
+      zcash_transaction.transparent_part().inputs.size(),
+      2 /* actions count for 1 orchard output no orchard inputs */));
+
+  // Pick orchard outputs
+  ZCashTransaction::OrchardOutput orchard_output;
+  auto addr_bytes =
+      zcash_wallet_service_->keyring_service()->GetOrchardRawBytes(
+          account_id_, mojom::ZCashKeyId::New(account_id_->account_index,
+                                              1 /* internal */, 0));
+  if (!addr_bytes) {
```

Comment:
`mojom::ZCashKeyId::New(account_id_->account_index, 1 /* internal */, 0)`
This most likely needs a seprate function with comment that we have only one internal(change) address for an account

Reply:
Done